core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -51,10 +51,14 @@ object ConfigSource {
     valueDelimiter: Option[Char]
   ): UIO[ConfigSource[String, String]] =
     UIO.effectTotal(argsAsKeyValues(args)).map { kvs =>
-      mergeAll(
-        kvs.map { kv =>
-          fromMapInternal(Map(kv))(splitDelimited(_, valueDelimiter), keyDelimiter, "command line arg")
-        }
+      ConfigSource.fromPropertyTrees(
```

Comment:
https://github.com/zio/zio-config/issues/248
you can get rid of UIO where modifying this method

Reply:
Yea sure